src/java.base/share/classes/java/lang/module/ModuleDescriptor.java

```diff
@@ -310,9 +310,8 @@ public int compareTo(Requires that) {
          */
         @Override
         public boolean equals(Object ob) {
-            if (!(ob instanceof Requires))
+            if (!(ob instanceof Requires that))
```

Comment:
it would be nicer if you inverted this to return (ob instanced Requires that) && ...
Same thing for the equals methods in Exports, Opens, Provides, ...

Reply:
I've refactored the various equals methods now, and they can be viewed in commit e9d9131